### PR TITLE
#984 Stopping NPE crash when no board selected

### PR DESCRIPTION
**Pull request by derimmer**

### CODE REVIEW

**Review by zgrose — COMMENTED** [review]
Per my note in the #984, seems like a reasonable quick fix assuming `testboardexists` is never null, too.

File: src/VASL/build/module/ASLMap.java [hunk 1]
```diff
@@ -175,6 +176,14 @@ public JPopupMenu getPopupMenu()
     @Override
     public synchronized void setBoards(Collection<Board> c) {
 
+        Iterator it = c.iterator();
+        while (it.hasNext()) {
```
Comment (uckelman) [inline src/VASL/build/module/ASLMap.java:180]: I would never use an `Iterator` for iterating over a whole collection; it's simpler to use a for-each loop:
```
for (VASLBoard testboardexists: c) {
  // do stuff
}
```
It's two lines shorter and involves no casting.

Comment (uckelman) [reply src/VASL/build/module/ASLMap.java:180]: Otherwise, this seems like a sensible solution to your problem.

**Comment by derimmer** [comment]
@uckelman @zgrose Thanks to you both for your comments here and on #984. I will proceed as per the suggested changes.

File: src/VASL/build/module/ASLMap.java (hunk 1 above)
Comment (derimmer) [reply src/VASL/build/module/ASLMap.java]: I have replaced the iterator with a for-each loop; no problem with that. 

However, for (VASLBoard testboardexists: c) { produces a type error as c is a collection of Board. So, I had to keep the cast. 

I have pushed up a new commit (see 5f9d734)

**Pull request merged**